### PR TITLE
Fix bug: Don't rename 'default' in `export { default as x } from "m";`

### LINKED ISSUE #22979: Renaming a default export breaks re-exports
**TypeScript Version:**  27.2, 2.8.1, typescript@2.9.0-dev.20180329

**Search Terms:** 
rename default export, rename re-export, rename reexport

**Code**

index.ts
```ts
export { default } from "./module1";
export { default as mod1 } from "./module1";
```

module1.ts
```ts
const module1 = {};
export default module1;
```

**Expected behavior:**
Renaming `module1` to `module2` in `module1.ts` doesn't impact `index.ts`.

**Actual behavior:**
Re-exports in `index.ts` are renamed as follows:

```ts
export { module2 } from "./module1";
export { module2 as mod1 } from "./module1";
```

### PR DESCRIPTION
**Pull request by andy-ms**
Fixes #22979

### CODE REVIEW

File: tests/cases/fourslash/renameReExportDefault.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+////import { default as b } from "./b";
+////import [|b|] from "./b";
+
+// @Filename: /b.ts
+////const [|b|] = 0;
+////export default [|b|];
+
+const [r0, r1, r2] = test.ranges();
+verify.renameLocations(r0, [r0]);
+verify.renameLocations([r1, r2], [r0, r1, r2]);
```
Comment (sheetalkamat) [inline tests/cases/fourslash/renameReExportDefault.ts]: seems like r0 shouldnt be part of this. It is not ```import { b} from``` syntax which would need rename.

Comment (andy-ms) [reply tests/cases/fourslash/renameReExportDefault.ts]: It doesn't strictly need a rename, but it seems appropriate that if you `export default foo` and `import foo`, you intend for these two to continue having the same name? `export default bar` and `import foo` will work at runtime, but it makes the code confusing if the symbol now has multiple names.

Comment (mhegazy) [reply tests/cases/fourslash/renameReExportDefault.ts]: so why not the rename the `b` in `export { default as b } from "./b"` as well?

Comment (andy-ms) [reply tests/cases/fourslash/renameReExportDefault.ts]: Good idea. :+1:

**Pull request merged**